source/server/listener_manager_impl.cc

```diff
@@ -281,6 +297,9 @@ void ListenerManagerImpl::drainListener(ListenerImplPtr&& listener) {
     std::lock_guard<std::mutex> guard(draining_listeners_lock_);
     draining_it = draining_listeners_.emplace(draining_listeners_.begin(), std::move(listener),
                                               workers_.size());
+    // Must be done under lock. Set is used so hot restart / multiple processes converge to a
+    // single value. Same below inside the lambda.
+    stats_.total_listeners_draining_.set(draining_listeners_.size());
```

Comment:
Yes, exactly. Otherwise during hot restart if we use inc()/dec() on the gauge we get double the listeners. This way it basically converges and the gauges remain roughly static between the two processes. (For those not using hot restart it's a NOP). We do the same thing with healthy hosts, # of clusters, etc.